Public field naming issues in method taking `self: Ref<...>` receiver

When using a public field of a struct in a method on this struct that takes a `Ref<>` receiver, it seems the field name is converted to PascalCase inside the method, but the struct itself declares it as Capital_snake_case.

Here is a minimal reproducer:
```
import "go:fmt"

struct Foo {
  pub bar_baz: string,
}

impl Foo {
  fn foo(self: Ref<Foo>) {
    fmt.Println(self.bar_baz)
  }
}

fn main() {
  let mut foo = Foo { bar_baz: "quux" }
  foo.foo()
}
```

The Go translation gives:
```go
package main

import "fmt"

type Foo struct {
	Bar_baz string
}

func (f Foo) String() string {
	return fmt.Sprintf("Foo { bar_baz: %v }", f.Bar_baz)
}

func (f *Foo) foo() {
	fmt.Println(f.BarBaz)
}

func main() {
	foo := Foo{Bar_baz: "quux"}
	foo.foo()
}
```

Which, as expected from the generated code, fails to compile:
```
➜ lis run
# pubfield
./main.go:14:16: f.BarBaz undefined (type *Foo has no field or method BarBaz)
```

Interestingly, changing the method receiver to `fn foo(self)` does not exhibit this issue and generates the `foo()` method correctly:
```go
func (f Foo) foo() {
	fmt.Println(f.Bar_baz)
}
```

--- a/crates/emit/src/go/expressions/access.rs
+++ b/crates/emit/src/go/expressions/access.rs
@@ -154,6 +154,7 @@ impl Emitter<'_> {
 
         let is_prelude_type = expression_ty
             .resolve()
+            .strip_refs()
             .get_qualified_id()
             .is_some_and(|id| id.starts_with(go_name::PRELUDE_PREFIX));
 

--- a/tests/spec/emit/types.rs
+++ b/tests/spec/emit/types.rs
@@ -1518,6 +1518,27 @@ fn test() {
 }
 
 #[test]
+fn ref_receiver_preserves_snake_case_in_public_field() {
+    let input = r#"
+struct Foo {
+  pub bar_baz: string,
+}
+
+impl Foo {
+  fn show(self: Ref<Foo>) -> string {
+    self.bar_baz
+  }
+}
+
+fn test() {
+  let mut foo = Foo { bar_baz: "quux" }
+  let _ = foo.show()
+}
+"#;
+    assert_emit_snapshot!(input);
+}
+
+#[test]
 fn ref_map() {
     let input = r#"
 fn test(m: Ref<Map<string, int>>) {

